primitives/beefy/src/commitment.rs

```diff
@@ -152,6 +156,115 @@ impl<TBlockNumber> SignedCommitment<TBlockNumber> {
 	}
 }
 
+/// Type to be used to denote placement of signatures
+type BitField = Vec<u8>;
+/// Compress 8 bit values into a single u8 Byte
+const CONTAINER_BIT_SIZE: usize = 8;
+
+/// Compressed representation of [`SignedCommitment`], used for encoding efficiency.
+#[derive(Clone, Debug, PartialEq, Eq, Encode, Decode)]
+struct CompactSignedCommitment<TCommitment> {
+	/// The commitment signatures, as is.
+	commitment: TCommitment,
+	/// A `Vec<u8>`, where each bit in a `u8` represents an `Option<Signature>`
+	/// in the uncompressed GRANDPA validators' signatures.
+	signatures_from: BitField,
+	/// Length of the `Vec` containing uncompressed GRANDPA validators' signatures.
+	signatures_len: u32,
+	/// A `Vec` containing all `Signature` values present in the uncompressed
+	/// GRANDPA validators' signatures.
+	signatures_compact: Vec<Signature>,
```

Comment:
```suggestion
	/// A `Vec` containing all `Signature`s present in the original [`SignedCommitment`].
	///
	/// Note that in order to associate a `Signature` from this `Vec` with a validator, one needs
	/// to look at the `signatures_from` bitfield, since some validators might have not produced a signature.
	signatures_compact: Vec<Signature>,
```